File: 2023-2/Clases/clase26_java03/SextoProyecto/Circulo.java
class Circulo{
	private String identificacion;
	private double radio;
	private final double PI = 3.141592;

	public Circulo(){
		this.identificacion = "C001";
		this.radio = 5;
	}

	public Circulo(String identificacion, double radio){
		this.identificacion = identificacion;
		this.radio = radio;
	}

	public void setIdentificacion(String identificacion){
		this.identificacion = identificacion;
	}

	public String getIdentificacion(){
		return this.identificacion;
	}

	public void setRadio(double radio){
		this.radio = radio;
	}

	public double getRadio(){
		return radio;
	}

	public double area(){
		return PI*radio*radio;
	}

	public double perimetro(){
		return 2*PI*radio;
	}
}

File: 2025-2/Clase17/Herencia/src/Circulo.cpp
//
// Created by Erasmo on 1/10/25.
//

#include "Circulo.hpp"
#define PI 3.141592
using namespace std;

double Circulo::area() {
    return PI * radio * radio;
}

double Circulo::perimetro() {
    return 2 * PI * radio;
}

void Circulo::mostrarResultados() {
    cout.precision(2);
    cout << fixed;
    cout << nombre << endl;
    cout << "Radio : " << setw(10) << radio << endl;
    cout << "Area : " << setw(10) << area() << endl;
    cout << "Perimetro : " << setw(10) << perimetro() << endl;

}

Circulo::Circulo() {
    radio = 10;
}
